### PR TITLE
CSHARP-2936: Support [Serializable] when targeting netstandard2.0

### PR DESCRIPTION
**Pull request by MikalaiMazurenka**
JIRA: https://jira.mongodb.org/browse/CSHARP-2936
EG: https://evergreen.mongodb.com/version/6074d8d23627e056aee4fc20

Note: there were several `[Serializable]` occurrences in Legacy, but I omitted them.

### CODE REVIEW

File: src/MongoDB.Driver.Core/MongoExecutionTimeoutException.cs [hunk 1]
```diff
@@ -79,7 +79,7 @@ public MongoExecutionTimeoutException(ConnectionId connectionId, string message,
             AddErrorLabelsFromCommandResult(this, result);
         }
 
-#if NET452
+#if !NETSTANDARD1_5
```
Comment (rstam) [inline src/MongoDB.Driver.Core/MongoExecutionTimeoutException.cs:82]: Add between lines 90-91:

    _result = (BsonDocument)info.GetValue("_result", typeof(BsonDocument));

Add between lines 112-113:

    #if !NETSTANDARD1_5
            /// <inheritdoc/>
            public override void GetObjectData(SerializationInfo info, StreamingContext context)
            {
                base.GetObjectData(info, context);
                info.AddValue("_result", _result);
            }
    #endif

**Review by rstam — CHANGES_REQUESTED** [review]
One small change requested.

Good job and nice attention to detail!

**Comment by rstam** [comment]
@jyemin is it OK to ignore Legacy in this PR?

**Comment by jyemin** [comment]
Sure as long as we're tracking it somewhere

File: src/MongoDB.Driver.Core/MongoExecutionTimeoutException.cs (hunk 1 above)
Comment (MikalaiMazurenka) [reply src/MongoDB.Driver.Core/MongoExecutionTimeoutException.cs:82]: Good catch! I've updated the unit test to contain the `_result` field as well.

**Review by MikalaiMazurenka — COMMENTED** [review]
Thanks for appreciation!
Ready for review

**Review by rstam — APPROVED** [review]
LGTM

**Pull request merged**